x-pack/plugins/security_solution/public/explore/components/risk_score/top_risk_score_contributors_alerts/index.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import { TableId } from '@kbn/securitysolution-data-table';
+import { EuiFlexGroup, EuiFlexItem, EuiPanel } from '@elastic/eui';
+import type { Filter } from '@kbn/es-query';
+
+import { HeaderSection } from '../../../../common/components/header_section';
+
+import * as i18n from './translations';
+import type { RiskInputs } from '../../../../../common/risk_engine';
+import { RiskScoreEntity } from '../../../../../common/risk_engine';
+import type { HostRiskScore, UserRiskScore } from '../../../../../common/search_strategy';
+import { ALERTS_TABLE_REGISTRY_CONFIG_IDS } from '../../../../../common/constants';
+import { AlertsTableComponent } from '../../../../detections/components/alerts_table';
+import { GroupedAlertsTable } from '../../../../detections/components/alerts_table/alerts_grouping';
+import { useGlobalTime } from '../../../../common/containers/use_global_time';
+import { useDeepEqualSelector } from '../../../../common/hooks/use_selector';
+import { inputsSelectors } from '../../../../common/store/inputs';
+import { useUserData } from '../../../../detections/components/user_info';
+import { useSourcererDataView } from '../../../../common/containers/sourcerer';
+import { SourcererScopeName } from '../../../../common/store/sourcerer/model';
+
+export interface TopRiskScoreContributorsAlertsProps {
+  toggleStatus: boolean;
+  toggleQuery?: (status: boolean) => void;
+  riskScore: HostRiskScore | UserRiskScore;
+  riskEntity: RiskScoreEntity;
+  loading: boolean;
+}
+
+export const TopRiskScoreContributorsAlerts: React.FC<TopRiskScoreContributorsAlertsProps> = ({
+  toggleStatus,
+  toggleQuery,
+  riskScore,
+  riskEntity,
+  loading,
+}) => {
+  const { to, from } = useGlobalTime();
+  const [{ loading: userInfoLoading, signalIndexName, hasIndexWrite, hasIndexMaintenance }] =
+    useUserData();
+  const { runtimeMappings } = useSourcererDataView(SourcererScopeName.detections);
+  const getGlobalFiltersQuerySelector = useMemo(
+    () => inputsSelectors.globalFiltersQuerySelector(),
+    []
+  );
+  const getGlobalQuerySelector = useMemo(() => inputsSelectors.globalQuerySelector(), []);
+
+  const query = useDeepEqualSelector(getGlobalQuerySelector);
+  const filters = useDeepEqualSelector(getGlobalFiltersQuerySelector);
+
+  const inputFilters = useMemo(() => {
+    const riskScoreEntity =
+      riskEntity === RiskScoreEntity.host
```

Comment:
nit: You could create a type guard function to avoid the cast. 
https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards